src/libraries/Common/tests/System/Security/Cryptography/MLKemContractTests.cs

```diff
@@ -0,0 +1,922 @@
+            // be written to because it is trimmed to the exact size, which ends up with the buffers beside each
+            // other, which should work.
+            byte[] buffer = new byte[algorithm.SharedSecretSizeInBytes + algorithm.CiphertextSizeInBytes + 10];
+            Memory<byte> sharedSecretBuffer = buffer.AsMemory(0, algorithm.SharedSecretSizeInBytes + 10);
+            Memory<byte> ciphertextBuffer = buffer.AsMemory(algorithm.SharedSecretSizeInBytes);
+            AssertExtensions.TrueExpression(sharedSecretBuffer.Span.Overlaps(ciphertextBuffer.Span));
+
+            using MLKemContract kem = new(algorithm)
+            {
+                OnEncapsulateCore = (Span<byte> ciphertext, Span<byte> sharedSecret) =>
+                {
+                    AssertSameBuffer(ciphertext, ciphertextBuffer.Span.Slice(0, algorithm.CiphertextSizeInBytes));
+                    AssertSameBuffer(sharedSecret, sharedSecretBuffer.Span.Slice(0, algorithm.SharedSecretSizeInBytes));
+                    AssertExtensions.FalseExpression(ciphertext.Overlaps(sharedSecret));
+                }
+            };
+
+            kem.Encapsulate(
+                ciphertextBuffer.Span,
+                sharedSecretBuffer.Span,
+                out int ciphertextWritten,
+                out int sharedSecretWritten);
+
+            Assert.Equal(algorithm.CiphertextSizeInBytes, ciphertextWritten);
+            Assert.Equal(algorithm.SharedSecretSizeInBytes, sharedSecretWritten);
+        }
+
+        [Fact]
+        public static void Encapsulate_Written_Disposed()
+        {
+            MLKemContract kem = new(MLKemAlgorithm.MLKem512);
+            kem.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => kem.Encapsulate(
+                new byte[kem.Algorithm.CiphertextSizeInBytes],
+                new byte[kem.Algorithm.SharedSecretSizeInBytes],
+                out _,
+                out _));
+        }
+
+        [Theory]
+        [MemberData(nameof(MLKemTestData.MLKemAlgorithms), MemberType = typeof(MLKemTestData))]
+        public static void Encapsulate_Allocated(MLKemAlgorithm algorithm)
+        {
+            using MLKemContract kem = new(algorithm)
+            {
+                OnEncapsulateCore = (Span<byte> ciphertext, Span<byte> sharedSecret) =>
+                {
+                    ciphertext.Fill(0xCA);
+                    sharedSecret.Fill(0xFE);
+                }
+            };
+
+            byte[] ciphertext = kem.Encapsulate(out byte[] sharedSecret);
+
+            Assert.Equal(algorithm.CiphertextSizeInBytes, ciphertext.Length);
+            Assert.Equal(algorithm.SharedSecretSizeInBytes, sharedSecret.Length);
+            AssertExtensions.FilledWith<byte>(0xCA, ciphertext);
+            AssertExtensions.FilledWith<byte>(0xFE, sharedSecret);
+        }
+
+        [Fact]
+        public static void Encapsulate_Allocated_Disposed()
```

Comment:
I'm fine with testing individually; but the high meta point is "once Dispose has run no Core methods at all are called, and everything throws ObjectDisposedException -- except ___".  Where ___ is "Dispose(), get_Algorithm, and ???" where ??? might be the empty set :smile:

So I'd probably do it as just one big "verify the state post-Dispose", but I'm talking out loud, not seeking change.